### ISSUE TITLE
FR: Create a new Material component for searching within mdSelect.

### ISSUE THREAD

**Issue by ErinCoughlan** [angular/material#7385]
Blocked by #7383 

Using the md-select-header, add an input box with default search behavior (case insensitive contains).

Create a new directive like `md-menu-filter` that people can drop into the menu header.

**Comment by EladBezalel** [comment]
@ErinCoughlan isn't that autocomplete?

**Comment by ErinCoughlan** [comment]
@EladBezalel It's a similar component. If we used autocomplete, we would need to add multi-select options along with using a custom template to get the rest of the select behavior.

Here's a screenshot of what I'm looking for (but a Material version):
![image](https://cloud.githubusercontent.com/assets/2660421/13479492/20397050-e08c-11e5-91e4-99000d04161a.png)

I'm hoping there is a way to combine the work for autocomplete, make it more generic to support multiple and single select, and then use it within both autocomplete and select since they are so similar.

**Comment by kathkroll** [comment]
The mdChips component when used with autocomplete seems to accomplish some of this functionality, but there are a fair amount of bugs with this combination. One notable one being that you can't remove used selections from the available dropdown without introducing another issue, as detailed [here](https://github.com/angular/material/issues/6674).

**Comment by ErinCoughlan** [comment]
@kathroll For a select component, you actually want selected items to appear in the search results. Consider a list of 10,000 items. You search for a value and select all of the items that appeared. You then clear the search term, notice you selected an incorrect item, and then have to scroll through 10,000 items to find and unselect it. That's a broken user experience.

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.
